src/System.Runtime/tests/System/Reflection/InvokeRefReturn.netcoreapp.cs

```diff
@@ -135,7 +135,7 @@ private static void TestRefReturnInvoke<T>(T value, Func<PropertyInfo, TestClass
             }
             else
             {
-                Assert.Same(value, rv);
+                Assert.Same((object)value, rv);
```

Comment:
feels like this pattern (if valuetype then equal else cast to object and do same) ought to be encapsulated in Assert somehow?